patches/minecraft/net/minecraft/client/renderer/entity/RenderItem.java.patch

```diff
@@ -9,7 +9,61 @@
          this.func_175041_b();
      }
  
-@@ -226,7 +226,7 @@
+@@ -153,7 +153,7 @@
+ 
+                 if (p_180454_1_.func_77962_s())
+                 {
+-                    this.func_180451_a(p_180454_2_);
++                    this.func_180451_a(p_180454_2_, p_180454_1_);
+                 }
+             }
+ 
+@@ -161,33 +161,18 @@
+         }
+     }
+ 
+-    private void func_180451_a(IBakedModel p_180451_1_)
++    private void func_180451_a(IBakedModel p_180451_1_, ItemStack stack)
+     {
+-        GlStateManager.func_179132_a(false);
```

Comment:
The original goal was to render the whole item list in one shot basically, caching a list of quads that could be dumped to the GPU in one call. Once it was proven, Forge could do something similar maybe. I ended up stopping short of that goal because of hangups like this, where special cases allow modders to interject their own OpenGL calls in the middle of the rendering pipeline, or rely on other things that can't be cached. For good rendering performance, I think this should be kept as simple as possible.

While it is possible for me to handle this case, I don't think it's a step in the right direction. Adding more power for modders is good, but it should be weighed against the costs on the other side like the ability to reason about and optimize the rendering code.